torch/testing/_internal/common_methods_invocations.py

```diff
@@ -474,6 +474,177 @@ def sample_inputs_linalg_det(op_info, device, dtype, requires_grad):
 
+def sample_inputs_linalg_det_singular(op_info, device, dtype, requires_grad):
+    kw = dict(device=device, dtype=dtype)
+    singular_det_cutoff = 1e-8
+
+    # Create a batch of randomized square singular matrices. Each matrix is
+    # generated by performing a matrix multiply of two random matrices of
+    # size (size[-1], rank) and (rank, size[-1])
+    def make_singular_matrix_batch_base(size, rank, device, dtype):
+        assert size[-1] == size[-2]
+        assert rank > 0 and rank <= size[-1]
+
+        batch = torch.empty(size, device=device, dtype=dtype)
+
+        matrix_size = size[-2:]
+        view_size = (int(np.prod(size[:-2])),) + matrix_size
+        batch_view = batch.view(view_size)
+
+        a_size = (size[-1], rank)
+        b_size = (rank, size[-1])
+
+        for matrix_idx in range(batch_view.size(0)):
+            num_tries = 0
+            max_tries = 50
+            is_matrix_acceptable = False
+
+            while not is_matrix_acceptable:
+                assert num_tries < max_tries, (
+                    f'could not create singular matrix of size {matrix_size}'
+                    f' after trying {max_tries} times')
+
+                lo = -10
+                hi = 10
+                if dtype.is_complex:
+                    a = torch.randint(lo, hi, a_size, device=device).to(dtype).add(
+                        1j * torch.randint(lo, hi, a_size, device=device).to(dtype))
+                    b = torch.randint(lo, hi, b_size, device=device).to(dtype).add(
+                        1j * torch.randint(lo, hi, b_size, device=device).to(dtype))
+
+                    # Avoiding zeros in a and b decreases likelihood of
+                    # producing an unstable gradient
+                    a[a.eq(0)] = torch.tensor(random.choice([hi, 1j * hi]), device=device, dtype=dtype)
+                    b[b.eq(0)] = torch.tensor(random.choice([hi, 1j * hi]), device=device, dtype=dtype)
+                else:
+                    a = torch.randint(lo, hi, a_size, device=device, dtype=dtype)
+                    b = torch.randint(lo, hi, b_size, device=device, dtype=dtype)
+                    a[a.eq(0)] = hi
+                    b[b.eq(0)] = hi
+
+                # NOTE: LU is less stable than the SVD used previously,
+                # hence we rescale the input for better stability
+                a /= 10
+                b /= 10
+
+                matrix = a @ b
+
+                # Ensure that the rank is actually what we wanted
+                if torch.matrix_rank(matrix) != rank:
+                    is_matrix_acceptable = False
+
+                # Ensure that linalg_det_backward will consider the matrix as singular
+                elif matrix.det().abs().gt(singular_det_cutoff):
+                    is_matrix_acceptable = False
+
+                elif (torch.device(device).type == 'cpu' and rank == 1) or (rank > 2):
+                    # TODO: linalg_det_backward can sometimes unstable if
```

Comment:
Did you try removing this check? I'm curious if it's still needed or not

Reply:
Good point!

Reply:
Removing it still makes things pass.